src/transforms/reduce/merge_strategy.rs

```diff
@@ -68,8 +68,8 @@ impl ReduceValueMerger for DiscardMerger {
         Ok(())
     }
 
-    fn insert_into(self: Box<Self>, k: KeyString, v: &mut LogEvent) -> Result<(), String> {
-        v.insert(event_path!(k.as_str()), self.v);
+    fn insert_into(self: Box<Self>, path: OwnedTargetPath, v: &mut LogEvent) -> Result<(), String> {
```

Comment:
To optimise this a bit - let's revert it back to KeyString and use here `parse_path_and_insert`? In this case we would parse paths only once when flushing reduced events.

Reply:
Should these be `impl TargetPath` or equivalent instead? I'm just wondering about avoiding requiring an owned path for callers that can use a borrowed one.

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/20800/commits/f566cb3c7a865f4b58660f5a1cc212e1b9c4464e